ontology/yaml/resources/ELECTRICAL/entity_types/UPS.yaml

```diff
@@ -83,3 +83,17 @@ UPS_SS_UPSBM:
   - UPS
   - SS
   - UPSBM
+
+UPS_UPSBM:
```

Comment:
You cannot name this canonical type using an abstract type you aren't implementing. You need to fully implement `UPSBM` instead. We want to avoid defining individual fields on a canonical type.

Based on the set of fields you have here, it seems like you'd want to implement `UPSBM` and `IOBM`

Reply:
is_canonical: false

I don't want to implement IOBM because it contains quite a few required fields that we don't have.  UPSBM requires "remaining_charge_sensor", which we also don't have.  I've created a field called "remaining_battery_time_sensor" that accurately reflects the data we do have.

Reply:
I've changed the type name to a non-canonical type name UPSTR